src/main/java/com/sphy/hotelmanagementapplication/service/HotelService.java

```diff
@@ -137,11 +148,13 @@ public String deleteHotel(Long id){
 	 * update a hotel
 	 * @param hotelDTO the hotel to be updated
 	 * @return the updated hotel
-	 * @throws NullPointerException
+	 * @throws ApiRequestException if the hotel does not exist
 	 */
-	public HotelDTO updateHotel(HotelDTO hotelDTO) throws Exception {
-		Optional<Hotel> hotelOpt = hotelRepository.findById(hotelDTO.getId());
-		if (hotelOpt.isPresent()){
+	public HotelDTO updateHotel(HotelDTO hotelDTO) throws ApiRequestException {
+
+		if (!hotelRepository.existsById(hotelDTO.getId())){
+			throw new ApiRequestException("The hotel with name: " + hotelDTO.getName() + " does not exist to update it");
```

Comment:
You check if a hotel exists by id, the error should indicate that the hotel with that id doesnt exist